### PR TITLE
update delay js default exclusions regex

### LINKED ISSUE #4371: jQuery regex excludes jQuery plugins
**Before submitting an issue please check that you’ve completed the following steps:**
- [x] Made sure you’re on the latest version `3.9.4`
- [x] Used the search feature to ensure that the bug hasn’t been reported before

**Describe the bug**
`/jquery-?[0-9.](.*)(.min|.slim|.slim.min)?.js` regex has false positives for `jquery.string.js`,which is widely used as naming convention for jQuery plugins. 

This can lead to excluding files that don't need exclusions, potentially creating dependency conflicts.

**To Reproduce**
1. Use `/jquery-?[0-9.](.*)(.min|.slim|.slim.min)?.js` for exclusions
2. Test against the following sample file names: 
```
jquery.js
jquery.min.js
jquery.slim.js
jquery.slim.min.js
jquery-123.min.js
jquery-123.slim.js
jquery-123.slim.min.js
jquery.123.min.js
jquery.123.slim.js
jquery.123.slim.min.js
jquery-skip.min.js
jquery-skip.slim.js
jquery-skip.slim.min.js
jquery.skip.min.js
jquery.skip.slim.js
jquery.skip.slim.min.js
```
3. See that all will return as a match, except for the `jquery-skip` ones.

**Expected behavior**
The REGEX should only match the jQuery library and skip anything else.

**Screenshots**
N/A

**Additional context**
Slack discussion - https://wp-media.slack.com/archives/C08N8J6VC/p1625757535079900

Maybe we could use 2 regexes to cover all cases? 

`/jquery(.min|.slim|.slim.min)?.js` will exclude all jQuery libraries without a version number on them.
`/jquery.*?([0-9]+)(.min|.slim|.slim.min)?.js` will exclude all jQuery libraries with a version number before the file extension. Avoids false positives if a plugin uses the version number as well. 

Managed to combine both regexes into one line: 

`/jquery((|.min|.slim|.slim.min)|.*?([0-9]+)(|.min|.slim|.slim.min))?.js`

Another variant that includes jquery-migrate as well: 

```
jquery(-migrate)?((|.min|.slim|.slim.min)|.*?([0-9]+)(|.min|.slim|.slim.min))?.js
```

It returns: 

```
jquery.js
jquery.min.js
jquery.slim.js
jquery.slim.min.js
jquery-123.min.js
jquery-123.slim.js
jquery-123.slim.min.js
jquery.123.min.js
jquery.123.slim.js
jquery.123.slim.min.js
```

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

### PR DESCRIPTION
**Pull request by mostafa-hisham**
## Description
update delay js default exclusions regex to handle false positives for the widely used naming convention for jQuery plugins.

Fixes #4371 

## Type of change

- [ ] Enhancement (non-breaking change which improves an existing functionality)
- [ ] This change requires a documentation update

## Is the solution different from the one proposed during the grooming?

No
## How Has This Been Tested?

- [ ] Unit and integration

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by remyperona — APPROVED** [review]

**Comment by piotrbak** [comment]
@CrochetFeve0251 @mostafa-hisham Is it ready for review, or we expect further changes here?

**Comment by piotrbak** [comment]
@wp-media/php Could you review one again the regex change here?

**Review by remyperona — APPROVED** [review]

File: inc/ThirdParty/Themes/MinimalistBlogger.php [hunk 1]
```diff
@@ -35,8 +35,7 @@ public static function get_subscribed_events() {
 	 * @return array
 	 */
 	public function exclude_jquery_from_delay_js( array $exclusions = [] ) {
-		$exclusions[] = '/jquery-?[0-9.](.*)(.min|.slim|.slim.min)?.js';
-		$exclusions[] = '/jquery-migrate(.min)?.js';
+		$exclusions[] = '/jquery(-migrate)?((|.min|.slim|.slim.min)|.*?([0-9]+)(|.min|.slim|.slim.min))?.js';
```
Comment (Mai-Saad) [inline inc/ThirdParty/Themes/MinimalistBlogger.php]: @mostafa-hisham I think exclusion here shall be `\/jquery(-migrate)?-?([0-9.]+)?(.min|.slim|.slim.min)?.js(\?(.*))?$`, what do you think? @Tabrisrp 
Note: the current pattern will match `/jquery.jsTest.js` which shouldn't be the case.

**Comment by piotrbak** [comment]
@Mai-Saad What kind of feedback is needed here?

**Comment by Mai-Saad** [comment]
@piotrbak recommended code change here https://github.com/wp-media/wp-rocket/pull/5869/files#diff-0fd71f0a6353f5d5dc816ec9361fdec2df328d7aeefc06b5222d1bb3c33d4905R38

**Review by Mai-Saad — APPROVED** [review]
Working as expected now.

Note: this change is only for UI not for the dynamic list

**Pull request merged**